src/main/deltachat/messagelist.js

```diff
@@ -56,16 +56,19 @@ module.exports = class DCMessageList extends SplitOut {
     const messages = []
     for (let i = 0; i < messageIdsToRender.length; i++) {
       const id = messageIdsToRender[i]
-      const json = this.messageIdToJson(id)
-      if (id === C.DC_MSG_ID_DAYMARKER && i > 0) {
-        if (messages[i - 1]) {
-          json.daymarker = {
-            timestamp: messages[i - 1].msg.timestamp,
-            id: 'd' + i
-          }
+      if (id !== C.DC_MSG_ID_DAYMARKER) {
```

Comment:
I don't know how generic this code is, but there are other special message IDs that could be returned, `DC_MSG_ID_MARKER1` being the obvious one, but perhaps you should check `id <= DC_MSG_ID_LAST_SPECIAL` instead?

Reply:
I am preparing a better fix: #1114